### ISSUE TITLE
issues with MNE study template

### ISSUE THREAD

**Issue by jasmainak** [mne-tools/mne-study-template#32]
Yesterday, with @larsoner we asked some workshop participants to try out the code here. Some of the limitations I observed:

- the config file is a bit hard to parse. If you want to try just one script, it's not clear which params in the config file must be touched or changed. Perhaps a more explicit import at the top of each file would solve this issue:

```py
from config import blah
```

- It assumes a certain structure to the data but it's not BIDS. This means there are many paths that need to be defined in the config. I think for people using the study template, we should make it as simple as possible -- they should just have to point to the root and then it should throw informative errors if certain files were expected, e.g., a list of participants, an events file etc. cc @sappelhoff 

- the code assumes fif file format. If we use, mne_bids.io.read_raw_bids, it should be able to triage over different files without making this assumption

**Comment by agramfort** [comment]
+1000

you put your finger on what the BIDS integration should fix. The config
should be become much
smaller.

by supporting more use cases while not breaking the old ones we will build
something solid.

**Comment by sappelhoff** [comment]
> from config import blah

+1 that'd be much more transparent. I could give that a shot.

> It assumes a certain structure to the data but it's not BIDS.

is the idea to make the mne-study-template rely on BIDS completely?

**Comment by jasmainak** [comment]
> is the idea to make the mne-study-template rely on BIDS completely?

I would say yes because all MEG data will hopefully be in BIDS format soon thanks to your GSoC :-)

I would start a WIP PR to see how far we can go with mne-bids

**Comment by jasmainak** [comment]
> +1 that'd be much more transparent. I could give that a shot.

go for it!

**Comment by agramfort** [comment]
+1 for make the study template work on BIDS formated data

>

**Comment by sappelhoff** [comment]
closing:

- mne-study-template is now BIDS compatible (at least, the first steps are done)
- there was substantial disagreement between several parties about how to deal with the imports - leaving this untouched until we have a solution that makes everybody happy (or everybody slightly less unhappy)
